plonky2_u32/src/gadgets/multiple_comparison.rs

```diff
@@ -0,0 +1,149 @@
+use plonky2::hash::hash_types::RichField;
+use plonky2::iop::target::{BoolTarget, Target};
+use plonky2::plonk::circuit_builder::CircuitBuilder;
+use plonky2_field::extension_field::Extendable;
+use plonky2_util::ceil_div_usize;
+
+use super::arithmetic_u32::U32Target;
+use crate::gates::comparison::ComparisonGate;
+
+/// Returns true if a is less than or equal to b, considered as base-`2^num_bits` limbs of a large value.
+/// This range-checks its inputs.
+pub fn list_le_circuit<F: RichField + Extendable<D>, const D: usize>(
```

Comment:
I wasn't too sure about where to put the comparison gadget and gate. It is currently only used in the biguint code, so it could also make sense to move it there.